src/DonationForms/Repositories/DonationFormDataRepository.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Give\DonationForms\Repositories;
+
+
+use Give\DonationForms\DonationFormDataQuery;
+use Give\DonationForms\Adapter\Form;
+use Give\DonationForms\ValueObjects\GoalType;
+
+class DonationFormDataRepository
+{
+    private array $amounts;
+    private array $subscriptionAmounts = [];
+    private array $donationsCount;
+    private array $subscriptionDonationsCount = [];
+    private array $donorsCount;
+    private array $subscriptionDonorsCount = [];
+
+    /**
+     * @param int[] $ids
+     *
+     * @return DonationFormDataRepository
+     */
+    public static function forms(array $ids): DonationFormDataRepository
+    {
+        $self = new self();
+
+        $donations = DonationFormDataQuery::donations($ids);
+
+        $self->amounts = $donations->collectIntendedAmounts();
+        $self->donationsCount = $donations->collectDonations();
+        $self->donorsCount = $donations->collectDonors();
+
+        // Set subscriptions data
+        if (defined('GIVE_RECURRING_VERSION')) {
+            $subscriptions = DonationFormDataQuery::subscriptions($ids);
+
+            $self->subscriptionAmounts = $subscriptions->collectInitialAmounts();
+            $self->subscriptionDonationsCount = $subscriptions->collectDonations();
+            $self->subscriptionDonorsCount = $subscriptions->collectDonors();
+        }
+
+        return $self;
+    }
+
+    /**
+     * @unreleased
+     *
+     * Get revenue for form
+     *
+     * @param Form $form
+     *
+     * @return int
+     */
+    public function getRevenue(Form $form): int
+    {
+        $data = $form->goalSettings->goalType->isSubscriptions()
+            ? $this->subscriptionAmounts
+            : $this->amounts;
+
+        foreach ($data as $row) {
+            if (isset($row['form_id']) && $row['form_id'] == $form->id) {
+                return (int)$row['sum'];
+            }
+        }
+
+        return 0;
```

Comment:
Remember, we don't want to cast this into a int to avoid php rounding float values.

```suggestion
     * @return int|float
     */
    public function getRevenue(Form $form)
    {
        $data = $form->goalSettings->goalType->isSubscriptions()
            ? $this->subscriptionAmounts
            : $this->amounts;

        foreach ($data as $row) {
            if (isset($row['form_id']) && $row['form_id'] == $form->id) {
                return $row['sum'];
            }
        }

        return 0;
```